06-building-a-program-start-to-finish/writing-tests/output/tests.js

```diff
@@ -0,0 +1,209 @@
+
+/* eslint-disable no-undef */
+
+const data = require('./data.json');
+const book = require('./books.json');
+const invalid = require('./invalid.json');
+const invalidContent = require('./invalid-content.json');
+
+const inverted = new InvertedIndex();
+
+describe('Inverted Index Test', () => {
+  describe('Validations', () => {
+    it('should return false if the JSON file is not valid', () => {
+      const validation = InvertedIndexUtility.isValidJson(invalid);
+      expect(validation).toBe(false);
+    });
+    it('should return false for invalid content', () => {
+      const validation = InvertedIndexUtility.isValidContent(invalidContent);
+      expect(validation).not.toBe(true);
+    });
+    it('should return false for an empty array', () => {
+      const validation = InvertedIndexUtility.isValidContent([]);
+      expect(validation).toBeFalsy();
+    });
+    it('should return true for a valid JSON file format', () => {
+      const validation = InvertedIndexUtility.isValidContent(data);
+      expect(validation).toBe(true);
+    });
+  });
+
+  describe('Get Index', () => {
+    it('should return false if index has not been created', () => {
+      const actual = inverted.getIndex('books.json');
+      expect(actual).toBe(false);
+    });
+  });
+
+  describe('Create Index', () => {
+    it('should return the index of the file', () => {
+      inverted.createIndex('data.json', data);
+      const indexCreated = inverted.getIndex('data.json').index;
+      const indexExpected = [
+        ['alice', true, false],
+        ['again', true, false],
+        ['harry', false, true],
+        ['lost', true, true],
+        ['is', true, true],
+        ['magic', false, true],
+        ['wand', false, true]
+      ];
+      expect(indexCreated).toEqual(indexExpected);
+    });
+  });
+
+  describe('Search Index', () => {
+    it('should return an empty array if no match is found', () => {
+      const query = 'you can\'t find me';
+      inverted.createIndex('data.json', data);
+      const actual = inverted.searchIndex(query, 'data.json').index;
+      const expected = [];
+      expect(actual).toEqual(expected);
+    });
+    it('should return the result if a query is found', () => {
+      const query = 'harry lost is magic';
+      inverted.createIndex('data.json', data);
+      const actual = inverted.searchIndex(query, 'data.json').index;
+      const expected = [
+        ['harry', false, true],
+        ['lost', true, true],
+        ['is', true, true],
+        ['magic', false, true]
+      ];
+      expect(actual).toEqual(expected);
+    });
+    it('should get search result for all files indexded', () => {
```

Comment:
Yes, there is. It checks if the result gotten from the search index method is equal to the expected variable. I had to create index before i search  though.